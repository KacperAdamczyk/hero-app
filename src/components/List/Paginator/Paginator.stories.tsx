/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import { action } from '@storybook/addon-actions';

import { Paginator } from './Paginator';

export default {
  title: 'List / Paginator',
  component: Paginator,
};

export const Default = () => (
  <div
    css={css`
      position: relative;
    `}
  >
    <div
      css={css`
        height: 75px;
        background-color: limegreen;
      `}
    ></div>
    <Paginator onLoadMore={action('onLoadMore')} />
  </div>
);
