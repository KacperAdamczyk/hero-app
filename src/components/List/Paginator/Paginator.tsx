import React, { FC } from 'react';
import styled from '@emotion/styled';

import { Button, ButtonColor } from 'components';

interface Props {
  onLoadMore: () => void;
}

const Container = styled.div`
  height: 75px;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  bottom: 0px;
  background: rgba(255, 255, 255, 0.75);
`;

export const Paginator: FC<Props> = ({ onLoadMore }) => (
  <Container>
    <Button color={ButtonColor.secondary} onClick={onLoadMore}>
      Load more
    </Button>
  </Container>
);
