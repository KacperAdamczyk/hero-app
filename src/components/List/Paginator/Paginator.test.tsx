import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Paginator } from './Paginator';

describe('Paginator', () => {
  test('should fire an event on click', () => {
    const onLoadMore = jest.fn();
    const { container, getByText } = render(
      <Paginator onLoadMore={onLoadMore} />,
    );

    fireEvent.click(getByText('Load more'));

    expect(onLoadMore).toHaveBeenCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
  });
});
