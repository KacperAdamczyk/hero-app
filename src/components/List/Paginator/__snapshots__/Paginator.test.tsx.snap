// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should fire an event on click 1`] = `
<div
  class="css-6duamk"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained css-1lu5or6-Button MuiButton-containedSecondary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Load more
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
